workflow/utils.py

```diff
@@ -0,0 +1,116 @@
+
+import datetime
+import os
+from pathlib import Path
+
+import pandas as pd
+
+# BIDS format
+BIDS_SUBJECT_PREFIX = 'sub-'
+BIDS_SESSION_PREFIX = 'ses-'
+
+# directory/file names
+DNAME_BACKUPS_MANIFEST = '.manifests'
+DNAME_BACKUPS_STATUS = '.doughnuts'
+FNAME_MANIFEST = 'mr_proc_manifest.csv'
+FNAME_STATUS = 'doughnut.csv'
+
+# for creating backups
+TIMESTAMP_FORMAT = '%Y%m%d_%H%M'
+EXT_SYMBOL = '.'
+SEP_FNAME_BACKUP = '-'
+
+# manifest file columns
+COL_SUBJECT_MANIFEST = 'participant_id'
+COL_BIDS_ID_MANIFEST = 'bids_id'
+COL_VISIT_MANIFEST = 'visit'
+COL_SESSION_MANIFEST = 'session'
+COL_DATATYPE_MANIFEST = 'datatype'
+COLS_MANIFEST = [COL_SUBJECT_MANIFEST, COL_BIDS_ID_MANIFEST, COL_VISIT_MANIFEST, 
+                 COL_SESSION_MANIFEST, COL_DATATYPE_MANIFEST]
+
+# status file columns
+COL_DICOM_DIR = 'participant_dicom_dir'
+COL_DICOM_ID = 'dicom_id'
+COL_DOWNLOAD_STATUS = 'downloaded'
+COL_ORG_STATUS = 'organized'
+COL_CONV_STATUS = 'converted'
+COLS_STATUS = [COL_SUBJECT_MANIFEST, COL_SESSION_MANIFEST, 
+               COL_DICOM_DIR, COL_DICOM_ID, COL_BIDS_ID_MANIFEST, 
+               COL_DOWNLOAD_STATUS, COL_ORG_STATUS, COL_CONV_STATUS]
+
+def participant_id_to_dicom_id(participant_id):
+    # keep only alphanumeric characters
+    participant_id = str(participant_id)
+    dicom_id = ''.join(filter(str.isalnum, participant_id))
+    return dicom_id
+
+def dicom_id_to_bids_id(dicom_id):
+    return subject_to_bids(dicom_id)
+
+def subject_to_bids(subject_id):
+    # add BIDS prefix if it doesn't already exist
+    subject_id = str(subject_id)
+    if subject_id.startswith(BIDS_SUBJECT_PREFIX):
+        return subject_id
+    else:
+        return f'{BIDS_SUBJECT_PREFIX}{subject_id}'
+
+def participant_id_to_bids_id(participant_id):
+    return dicom_id_to_bids_id(participant_id_to_dicom_id(participant_id))
```

Comment:
I think having both of these is a bit confusing in terms of whats `subject` and what's `participant`. 
We should either go with direct name transformations or the chained ones that forces participant_id --> dicom_id --> bids_id.